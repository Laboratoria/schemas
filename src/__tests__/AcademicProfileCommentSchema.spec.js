const mongoose = require('mongoose/browser');
const {
  AcademicProfileCommentSchema,
  UserSchema,
  AcademicProfileSchema,
} = require('../../')(mongoose);

describe('AcademicProfileEndorsmentSchema', () => {
  it('should fail validation when fields missing', () => {
    const doc = new mongoose.Document({}, AcademicProfileCommentSchema);
    expect(doc.validateSync().errors).toMatchSnapshot();
  });

  it('should validate example', (done) => {
    const academicProfile = new mongoose.Document({}, AcademicProfileSchema);
    const createdBy = new mongoose.Document({}, UserSchema);

    const doc = new mongoose.Document({
      academicProfile: academicProfile._id,
      createdBy: createdBy._id,
      type: 'tech',
      text: 'foo',
    }, AcademicProfileCommentSchema);

    doc.validate((err) => {
      expect(err).toBe(null);
      done();
    });
  });
});
