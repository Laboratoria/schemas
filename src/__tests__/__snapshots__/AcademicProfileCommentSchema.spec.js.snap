// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AcademicProfileEndorsmentSchema should fail validation when fields missing 1`] = `
Object {
  "academicProfile": [ValidatorError: Path \`academicProfile\` is required.],
  "createdBy": [ValidatorError: Path \`createdBy\` is required.],
  "type": [ValidatorError: Path \`type\` is required.],
}
`;
